OpenRA.Mods.Common/Traits/Sound/DeathSounds.cs

```diff
@@ -42,8 +43,8 @@ public void Killed(Actor self, AttackInfo e)
 
 			var cp = self.CenterPosition;
 
-			if (info.DeathTypes.Contains(e.Warhead.DeathType) ||
-				(!info.DeathTypes.Any() && !self.Info.Traits.WithInterface<DeathSoundsInfo>().Any(dsi => dsi.DeathTypes.Contains(e.Warhead.DeathType))))
+			if (info.DeathTypes.Any(x => e.Warhead.DamageTypes.Contains(x)) ||
```

Comment:
This is a monster condition.  Please pull out whatever temporary variables you need to make it readable, and you probably want to use `.Intersect()` instead of `.Any(.Contains())`.